src/dotnet-scaffolding/dotnet-scaffold-aspnet/ScaffoldSteps/ValidateEfControllerStep.cs

```diff
@@ -0,0 +1,204 @@
+
+    public ValidateEfControllerStep(
+        IFileSystem fileSystem,
+        ILogger<ValidateEfControllerStep> logger)
+    {
+        _fileSystem = fileSystem;
+        _logger = logger;
+    }
+
+    public override async Task<bool> ExecuteAsync(ScaffolderContext context, CancellationToken cancellationToken = default)
+    {
+        var efControllerSettings = ValidateEfControllerSettings();
+        var codeModifierProperties = new Dictionary<string, string>();
+        if (efControllerSettings is null)
+        {
+            return false;
+        }
+        else
+        {
+            context.Properties.Add(nameof(EfControllerSettings), efControllerSettings);
+        }
+
+        //initialize CrudControllerModel
+        _logger.LogInformation("Initializing scaffolding model...");
+        var efControllerModel = await GetEfControllerModelAsync(efControllerSettings);
+        if (efControllerModel is null)
+        {
+            _logger.LogError("An error occurred.");
+            return false;
+        }
+        else
+        {
+            context.Properties.Add(nameof(EfControllerModel), efControllerModel);
+        }
+
+        //Install packages and add a DbContext (if needed)
+        if (efControllerModel.DbContextInfo.EfScenario)
+        {
+            var dbContextProperties = AspNetDbContextHelper.GetDbContextProperties(efControllerSettings.Project, efControllerModel.DbContextInfo);
+            if (dbContextProperties is not null)
+            {
+                context.Properties.Add(nameof(DbContextProperties), dbContextProperties);
+            }
+
+            var projectBasePath = Path.GetDirectoryName(efControllerSettings.Project);
+            if (!string.IsNullOrEmpty(projectBasePath))
+            {
+                context.Properties.Add(Constants.StepConstants.BaseProjectPath, projectBasePath);
+            }
+
+            codeModifierProperties = AspNetDbContextHelper.GetDbContextCodeModifierProperties(efControllerModel.DbContextInfo);
+        }
+
+        context.Properties.Add(Constants.StepConstants.CodeModifierProperties, codeModifierProperties);
+        return true;
+    }
+
+    private EfControllerSettings? ValidateEfControllerSettings()
+    {
+        if (string.IsNullOrEmpty(Project) || !_fileSystem.FileExists(Project))
+        {
+            _logger.LogError("Missing/Invalid --project option.");
+            return null;
+        }
+
+        if (string.IsNullOrEmpty(Model))
+        {
+            _logger.LogError("Missing/Invalid --model option.");
+            return null;
+        }
+
+        if (string.IsNullOrEmpty(ControllerName))
+        {
+            _logger.LogError("Missing/Invalid --controller option.");
+            return null;
+        }
+        else
+        {
+            ControllerName = Path.GetFileNameWithoutExtension(ControllerName);
+        }
+
+        if (string.IsNullOrEmpty(ControllerType))
+        {
+            return null;
```

Comment:
Don't you want to log anything if `ControllerType` is null (so you return null)? You do it for the other cases, above

Reply:
oops should be logging, will add a simple error. Since we are initializing this property, it'll be internal like message (and we always initialize this)